clockwork.go

```diff
@@ -227,21 +213,45 @@ func (fc *fakeClock) Advance(d time.Duration) {
 }
 
 // BlockUntil blocks until the fakeClock has the given number of waiters.
+//
+// Deprecated: Prefer BlockUntilContext.
 func (fc *fakeClock) BlockUntil(n int) {
+	b := fc.newBlocker(n)
+	if b == nil {
+		return
+	}
+	<-b.ch
+}
+
+// BlockUntilContext blocks until the fakeClock has the given number of waiters
+// or the context is cancelled.
+func (fc *fakeClock) BlockUntilContext(ctx context.Context, n int) error {
+	b := fc.newBlocker(n)
+	if b == nil {
+		return nil
+	}
```

Comment:
```suggestion
	}

```

nit: I like better separated code blocks

Reply:
Done and I'll keep that in mind down the road.